dss/stepfunctions/visitation/implementation.py

```diff
@@ -107,7 +107,9 @@ def walker_failed(event, context, branch):
         "BackoffRate": 1
     },
     {
-        "ErrorEquals": ["States.Timeout"],
+        "ErrorEquals": ["States.Timeout",
+                        "States.TaskFailed",
+                        "Lambda.SdkClientException"],
```

Comment:
`States.TaskFailed` likely covers failure scenarios where we don't want to retry. Maybe we can try catching specific exceptions first.

Are you able to verify that `Lambda.SdkClientException` is succesfully caught? Also, consider including other observed Lambda exceptions, e.g. https://github.com/HumanCellAtlas/data-store/issues/1391#issuecomment-403161225

Reply:
I'm unable to verify the fix since we are unable to consistently reproduce the problem. I'll add in the additional errors and take out `States.TastFailed`